### PR TITLE
[BAHIR-144] Add flink-library-siddhi from Flink

### PR DESCRIPTION
**Pull request by haoch**
Moved from https://github.com/apache/flink/pull/2487 as suggested by Flink community.

https://issues.apache.org/jira/browse/BAHIR-144

### CODE REVIEW

**Comment by rmetzger** [comment]
Thanks a lot for moving the PR over here!

File: flink-library-siddhi/pom.xml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            <exclusions>
+                <exclusion>  <!-- declare the exclusion here -->
+                    <groupId>org.apache.directory.jdbm</groupId>
+                    <artifactId>apacheds-jdbm1</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.siddhi</groupId>
+            <artifactId>siddhi-query-api</artifactId>
```
Comment (rmetzger) [inline flink-library-siddhi/pom.xml]: Dependency is Apache 2.0 licensed --> good.

**Comment by rmetzger** [comment]
I think the change is good to be merged.
The tests might be unstable, but maybe that's just a local issue...we will see.
I think we should then soon add documentation for Siddhi as well.

I'm going to keep the PR open for a few more days in case somebody else wants to take a look.

File: flink-library-siddhi/src/test/java/org/apache/flink/streaming/siddhi/SiddhiCEPITCase.java [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+        );
+
+        DataStream<Event> output = SiddhiCEP
+            .define("inputStream", input, "id", "name", "price")
+            .cql("from inputStream insert into  outputStream")
+            .returns("outputStream", Event.class);
+        String path = tempFolder.newFile().toURI().toString();
+        output.print();
+        env.execute();
+        // Assert.assertEquals(6, getLineCount(path));
```
[the remark this answers is not in the document]
Comment (haoch) [reply flink-library-siddhi/src/test/java/org/apache/flink/streaming/siddhi/SiddhiCEPITCase.java]: Resolved

**Comment by haoch** [comment]
@rmetzger thanks for the review. As to documentation, I will continue to add more java doc and contribute the web site later.

BTW: What's bahir's site repo?

**Comment by haoch** [comment]
@rmetzger what's the plan or additional requirements for merging it?

**Comment by rmetzger** [comment]
I'm going to merge this pull request now. Sorry for the delay.

**Comment by haoch** [comment]
Thanks very much!

**Pull request closed without merging**